src/lib/number.jl

```diff
@@ -1,32 +1,63 @@
 
 @nograd floor, ceil, trunc, round, div
 
-@adjoint Base.literal_pow(::typeof(^), x::Number, ::Val{p}) where {p} =
-  Base.literal_pow(^,x,Val(p)),
-  Δ -> (nothing, Δ * conj(p * Base.literal_pow(^,x,Val(p-1))), nothing)
+function ChainRulesCore.rrule(
+    ::ZygoteRuleConfig, ::typeof(Base.literal_pow), ::typeof(^), x::Number, ::Val{p}
+) where {p}
+    function literal_pow_pullback(Δ)
+        dx = Δ * conj(p * Base.literal_pow(^,x,Val(p-1)))
+        return (NoTangent(), NoTangent(), dx, NoTangent())
+    end
+    return Base.literal_pow(^,x,Val(p)), literal_pow_pullback
+end
 
-@adjoint Base.convert(T::Type{<:Real}, x::Real) = convert(T, x), ȳ -> (nothing, ȳ)
```

Comment:
This is the only functional change, because
```julia
convert(::Type{T}, x::Number) where {T<:Number} = T(x)
```
and so the rule below should be sufficient

Reply:
never mind, breaks at second order in NNlib.jl